server/src/services/tooljet_db_bulk_upload.service.ts

```diff
@@ -157,6 +168,19 @@ export class TooljetDbBulkUploadService {
       const columnsInCsv = Object.keys(row);
       const transformedRow = columnsInCsv.reduce((result, columnInCsv) => {
         const columnDetails = internalTableColumnSchema.find((colDetails) => colDetails.column_name === columnInCsv);
+
+        const { keytype, data_type } = columnDetails;
+
+        if (keytype === 'PRIMARY KEY' && data_type !== 'serial') {
```

Comment:
I don't think we get `data_type` as serial explicitly. We might have to look at the default value and infer that it is of serial type. [[related](https://github.com/ToolJet/ToolJet/pull/9279#discussion_r1552132732)]